src/writing-policies/spec/06-signature-verifier-policies.md

```diff
@@ -0,0 +1,147 @@
+standard. This allows to implement a Secure Supply Chain for your cluster.
+
+Part of that is to ensure that all container images run the in the cluster are
+signed and verified, proving that they come from their stated authors, and
+haven't been tampered with (for other concerns, see the docs on [how we implement a Secure Supply Chain for the policies themselves](TODO link to https://github.com/kubewarden/docs/pull/100/files)).
+
+Sigstore signatures of containers are stored as layers of the container image.
+Policies that want to check Sigstore signatures of containers need then to check
+those container layers, and would need to pull the signature layers to see the
+signatures themselves.
+
+Obtaining and operating with those container layers needs to happen outside of
+the waPC guest (the policy), and on the waPC host (Kubewarden's
+`policy-server`).
+
+Just like for context-aware policies, the guest-host intercommunication is
+performed using the regular waPC host calling mechanism, and any guest
+implementing the waPC intercommunication mechanism is able to request this
+information.
+
+The different language SDKs for Kubewarden policies take care of all that, and
+provide functions for container image verification. This ensures a safe and
+tested codepath for verification. In addition, pulling images and
+cryptographically verifying signatures can be time and computationally
+expensive, so the SDKs stack ensures that both the image pulls and the
+verification computations are cached.
+
+The SDKs provide functions similar to the following:
+- ```
+  verify_pub_keys_image(
+      image_url: string,
+      vector_of_pub_keys: vector<string>,
+      vector_of_sigstore_annotations: Vector<(key, value: string)>
+      )
+      returns (is_trusted: bool, digest_of_verified_image: string)
+  ```
+- ```
+  verify_keyless_exact_match(
+      image_url: string,
+      vector_of_tuples_issuer_and_subject: vector<(issuer, subject: string)>,
+      vector_of_sigstore_annotations: vector<(key, value: string)>
+      )
+      returns (is_trusted: bool, digest_of_verified_image: string)
+  ```
+
+Both functions verify that the image is signed and satisfy the passed
+constraints.
+
+Note well: on success, the functions return the digest of the verified image. It
+is now on the policy to ensure that containers are instantiated from that
+digest, and not from tag that may or may not match that checksum digest (and
+therefore be compromised).
+
+
+## A concrete example
+
+The Kubewarden team [provides a verifier policy](TODO link to hub.kubewarden.io for the policy)
+that enforces Sigstore signatures for all containers, built on Rust and with the
+Rust SDK. The policy ensures that the containers are signed, and optionally,
+mutates the requests substituting the container tag for the exact checksum
+of the image matching the verified signature. It has also quality of life
+additions on the UX, see its docs for specifics.
```

Comment:
This is a personal taste, but I don't like the "It has quality of life"... statement. I would just say something like "check the policy documentation for more details about its settings"

Reply:
done in https://github.com/kubewarden/docs/pull/109/commits/7b6920fa261a7c92632e084b3532db7215f61ada!